src/material/core/datetime/date-adapter.ts

```diff
@@ -203,6 +203,14 @@ export abstract class DateAdapter<D> {
    */
   abstract invalid(): D;
 
+  /**
```

Comment:
In addition to `@param` and `@returns`, this method should have a description, something like
```ts
 /**
  * Given a potential date object, returns that same date object if it is
  * a valid date, or `null` if it's not a valid date.
  * @param ...
  * @returns ...
  */
```